ibm/resource_ibm_cbr_zone.go

```diff
@@ -0,0 +1,682 @@
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The href link to the resource.",
+			},
+			"created_at": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time the resource was created.",
+			},
+			"created_by_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "IAM ID of the user or service which created the resource.",
+			},
+			"last_modified_at": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The last time the resource was modified.",
+			},
+			"last_modified_by_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "IAM ID of the user or service which modified the resource.",
+			},
+			"version": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIBMCbrZoneValidator() *ResourceValidator {
+	validateSchema := make([]ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 "name",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^[a-zA-Z0-9 \\-_]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+		ValidateSchema{
+			Identifier:                 "account_id",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^[a-zA-Z0-9\\-]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+		ValidateSchema{
+			Identifier:                 "description",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^[\x20-\xFE]*$`,
+			MinValueLength:             0,
+			MaxValueLength:             300,
+		},
+		ValidateSchema{
+			Identifier:                 "transaction_id",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^[a-zA-Z0-9\-_]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+	)
+
+	resourceValidator := ResourceValidator{ResourceName: "ibm_cbr_zone", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMCbrZoneCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	contextBasedRestrictionsClient, err := meta.(ClientSession).ContextBasedRestrictionsV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createZoneOptions := &contextbasedrestrictionsv1.CreateZoneOptions{}
+
+	if _, ok := d.GetOk("name"); ok {
+		createZoneOptions.SetName(d.Get("name").(string))
+	}
+	if _, ok := d.GetOk("account_id"); ok {
```

Comment:
Instead of getting from user, derive it using BluemixUserDetails client.. For eg
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_iam_api_key.go#L122

Reply:
is this change mandatory ?

Reply:
done.